prompt_toolkit/selection.py

```diff
@@ -42,6 +42,10 @@ def __init__(self, original_cursor_position: int = 0,
 
         self.original_cursor_position = original_cursor_position
         self.type = type
+        self.shift_mode = False
```

Comment:
I think this attribute needs to go into ``key_bindings.emacs_state.EmacsState``, because it's specific to the Emacs key bindings.

Reply:
That's what I had initially done, but then you need to make sure that that flag is set back to False when the selection is removed by any existing or future keybinding. So either that means adding stuff in quite a few key bindings, or handling this in buffer.py, wherever selection_state is set to None (in exit_selection and copy_selection and maybe somewhere else). That's why I prefer having the shift_mode flag in selection_state, since it then becomes automatic. Also, having it in selection_state opens the way to extending the shift-selection mode to maybe the vi mode (or any other future mode?) in the future? See also my two points at the top of my initial pull request description. What do you think?

Reply:
That makes sense. Thanks! Just keep it.